### DISCUSSION TITLE
Wireframes

### DISCUSSION THREAD

**Discussion by DotNetDublin** [BlazingSaddlesSolutions/TimeKeeper#62] [Show and tell]
As mentioned in the weekly [roundup](https://github.com/BlazingSolutions/TimeKeeper/discussions/60),  one of the goals for this week is to mock additional features for the time entry page.

To that end I've created a [wireframes](https://github.com/BlazingSolutions/TimeKeeper/tree/wireframes) branch which we can use to wireframe not only additional functionality for the time entry page but all pages of the application going forward.

See below for details and as always suggestions are welcome.

This branch contains the project TimeKeeper.Wireframes which is a stand alone server side Blazor project.

The landing page of TimeKeeper.Wireframes shows a table of wire frames showing ID, Page Name, Status plus a link to GitHub which I was thinking would either be an issue or discussion relating to the page in question.

![Wireframes](https://user-images.githubusercontent.com/52583820/119421558-dff0a980-bcf6-11eb-9487-c003fbf9ccc5.jpg)

The below shows the wireframe for the time entry page and as you can see I've added functionality to allow the user to navigate days as well as displaying the total number of hours per week. If you look at the code on [TimeEntry.razor](https://github.com/BlazingSolutions/TimeKeeper/blob/wireframes/TimeKeeper.Wireframes/Pages/TimeEntry.razor) you'll see that the C# code is minimal and is only in place where it makes it easier to demonstrate the intent of a function on the page. It is not intended to guide how the function would operate when implemented if that makes sense?

![TimeEntry](https://user-images.githubusercontent.com/52583820/119421554-df581300-bcf6-11eb-98d9-15d56e19521e.jpg)

When working with the wireframes branch you should set TimeKeeper.Wireframes to be the sole startup project

![TimeKeeper Wireframes](https://user-images.githubusercontent.com/52583820/119421810-79b85680-bcf7-11eb-963c-ed883ad717b3.jpg)

![StartupProject](https://user-images.githubusercontent.com/52583820/119421808-791fc000-bcf7-11eb-8a11-92f87a573b3d.jpg)

**Comment by harperjohn** [comment]
I pulled and successfully ran it! Looks Great!

I just read this:
https://www.mockplus.com/blog/post/what-is-wireframe-in-software-development

Does this capture the essence of your vision of wireframes?

**Reply by DotNetDublin** [reply]
Yes and you could argue that we are producing dynamic wireframes although the article does state the below which probably won't be true in our case. By that I mean when the wireframe is implemented the user interface will probably look the exact same as the wireframe with the difference being the wiring (coding) that makes the functions represented in the wireframe work. Hope I'm making sense there?

> Nevertheless, we are still in the conception phase even with dynamic wireframes. The design only plays a minor role. Individual graphics elements are only presented in a simplified way since...

**Reply by harperjohn** [reply]
Interesting @DotNetDublin ! I like this... and I am (ironically) at this point with my "day job" project. The actual "wiring" is pretty complicated, but I did want to give the users a "good mockup" for trying/reviewing/updating cycles.

**Comment by DotNetDublin** [comment]
I think issue #13 shows how the flow of development could run nicely from user story to wireframes to coding.
